ginivision/src/main/java/net/gini/android/vision/internal/fileimport/FileChooserActivity.java

```diff
@@ -0,0 +1,243 @@
+
+    public static final int RESULT_ERROR = RESULT_FIRST_USER + 1;
+    public static final String EXTRA_OUT_ERROR = "GV_EXTRA_OUT_ERROR";
+
+    public static final int GRID_SPAN_COUNT = 4;
+
+    private RelativeLayout mLayoutRoot;
+    private RecyclerView mFileProvidersView;
+
+    public static boolean canChooseFiles(@NonNull final Context context) {
+        final List<ResolveInfo> imagePickerResolveInfos = queryImagePickers(context);
+        final List<ResolveInfo> imageProviderResolveInfos = queryImageProviders(context);
+        final List<ResolveInfo> pdfProviderResolveInfos = queryPdfProviders(context);
+
+        return imagePickerResolveInfos.size() > 0
+                || imageProviderResolveInfos.size() > 0
+                || pdfProviderResolveInfos.size() > 0;
+    }
+
+    public static Intent createIntent(final Context context) {
+        return new Intent(context, FileChooserActivity.class);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.gv_activity_file_chooser);
+        bindViews();
+        setupFileProvidersView();
+        overridePendingTransition(0, 0);
+    }
+
+    private void bindViews() {
+        mLayoutRoot = findViewById(R.id.gv_layout_root);
+        mFileProvidersView = findViewById(R.id.gv_file_providers);
+    }
+
+    private void setupFileProvidersView() {
+        mFileProvidersView.setLayoutManager(new GridLayoutManager(this, GRID_SPAN_COUNT));
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        populateFileProviders();
+        showFileProviders();
+    }
+
+    private void showFileProviders() {
+        mLayoutRoot.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                AutoTransition transition = new AutoTransition();
+                transition.setDuration(200);
+                TransitionManager.beginDelayedTransition(mLayoutRoot, transition);
+                RelativeLayout.LayoutParams layoutParams =
+                        (RelativeLayout.LayoutParams) mFileProvidersView.getLayoutParams();
+                layoutParams.addRule(RelativeLayout.BELOW);
+                layoutParams.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
+                mFileProvidersView.setLayoutParams(layoutParams);
+            }
+        }, 300);
+    }
+
+    @Override
+    public void onBackPressed() {
+        overridePendingTransition(0, 0);
+        hideFileProviders(new TransitionListenerAdapter() {
+            @Override
+            public void onTransitionEnd(@NonNull final Transition transition) {
+                FileChooserActivity.super.onBackPressed();
+            }
+        });
+
+    }
+
+    private void hideFileProviders(
+            @NonNull final Transition.TransitionListener transitionListener) {
+        AutoTransition transition = new AutoTransition();
+        transition.setDuration(200);
```

Comment:
this should be a constant and also be replaced [here](https://github.com/gini/gini-vision-lib-android/pull/78/files#diff-619a1c7ff56da2034a773a5b8c2c65f7R93)